pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -2044,23 +2097,99 @@ func DeleteProductServices(envName, productName string, serviceNames []string, l
 		log.Errorf("find product error: %v", err)
 		return err
 	}
+	if getProjectType(productName) == setting.HelmDeployType {
+		return deleteHelmProductServices(productInfo, serviceNames, log)
+	} else {
+		return deleteK8sProductServices(productInfo, serviceNames, log)
+	}
+}
+
+func deleteHelmProductServices(productInfo *commonmodels.Product, serviceNames []string, log *zap.SugaredLogger) error {
+	// init helm client
+	restConfig, err := kube.GetRESTConfig(productInfo.ClusterID)
+	if err != nil {
+		return e.ErrUpdateEnv.AddErr(err)
+	}
+	helmClient, err := helmtool.NewClientFromRestConf(restConfig, productInfo.Namespace)
+	if err != nil {
+		return e.ErrUpdateEnv.AddErr(err)
+	}
+
+	deleteServiceSet := sets.NewString(serviceNames...)
+	for serviceGroupIndex, serviceGroup := range productInfo.Services {
+		var group []*commonmodels.ProductService
+		for _, service := range serviceGroup {
+			if !deleteServiceSet.Has(service.ServiceName) {
+				group = append(group, service)
+			}
+		}
+		err := commonrepo.NewProductColl().UpdateGroup(productInfo.EnvName, productInfo.ProductName, serviceGroupIndex, group)
+		if err != nil {
+			log.Errorf("update product error: %v", err)
+			return err
+		}
+	}
+	renderset, err := commonrepo.NewRenderSetColl().Find(&commonrepo.RenderSetFindOption{
+		Name: productInfo.Namespace,
+	})
+	if err != nil {
+		log.Errorf("get renderSet error: %v", err)
+		return err
+	}
+	rcs := make([]*template.RenderChart, 0)
+	for _, v := range renderset.ChartInfos {
+		if !deleteServiceSet.Has(v.ServiceName) {
+			rcs = append(rcs, v)
+		}
+	}
+	renderset.ChartInfos = rcs
+
+	// create new renderset
+	if err := commonservice.CreateHelmRenderSet(renderset, log); err != nil {
+		log.Error("failed to create renderset, name %s, err: %s", renderset.Name, err)
+		return e.ErrUpdateEnv.AddErr(err)
+	}
 
+	productInfo.Render.Revision = renderset.Revision
+	err = commonrepo.NewProductColl().UpdateRender(renderset.EnvName, productInfo.ProductName, productInfo.Render)
+	if err != nil {
+		log.Errorf("failed to update product render info, renderName: %s, err: %s", productInfo.Render.Name, err)
+		return e.ErrUpdateEnv.AddErr(err)
+	}
+
+	// uninstall services
+	for _, service := range deleteServiceSet.List() {
+		go func(namespace, serviceName string) {
```

Comment:
If the deletion fails, does it affect the creation of new services with the same name?
At the product level, do users need to be aware of deletion exceptions?